toolkit/geo-compose/src/main/java/com/arcgismaps/toolkit/geocompose/GeoComposeState.kt

```diff
@@ -17,11 +17,42 @@
 
 package com.arcgismaps.toolkit.geocompose
 
+import kotlinx.coroutines.channels.BufferOverflow
+import kotlinx.coroutines.flow.MutableSharedFlow
+import kotlinx.coroutines.flow.SharedFlow
+import kotlinx.coroutines.flow.asSharedFlow
+
 /**
  * Represents the state for the GeoCompose.
  *
  * @since 200.3.0
  */
 public sealed class GeoComposeState {
-    // TODO add implementation
+
+    /**
+     * A [MutableSharedFlow] backing the public immutable viewpointChanged [SharedFlow].
+     *
+     * @since 200.3.0
+     */
+    private val _viewpointChanged: MutableSharedFlow<Unit> =
+        MutableSharedFlow(
+            extraBufferCapacity = Int.MAX_VALUE,
+            onBufferOverflow = BufferOverflow.DROP_OLDEST
+        )
+
+    /**
+     * A [SharedFlow] which notifies when the viewpoint of GeoCompose has changed.
```

Comment:
I think we might be better off to use [GeoView] instead of `GeoCompose`, which is not a real thing:
```suggestion
     * A [SharedFlow] which notifies when the viewpoint of the corresponding [GeoView] has changed.
```
Note sure, worth to discuss..

Reply:
Would it be better if we say `GeoComposeState` instead ?
I was thinking we don't want to refer to the underlying MapView/GeoView and restrict the conversation to the compose `Map`. So here I was wording it to state ... the viewpoint (property) of GeoComposeState interface has changed, but mistyped it as GeoCompose.